cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -501,33 +506,225 @@ __inline__ __device__ std::pair<char const*, char const*> trim_whitespaces_quote
+{
+  return to_duration<T>(begin, end);
+}
+
+// The purpose of these is merely to allow compilation ONLY
+template <>
+__inline__ __device__ cudf::string_view decode_value(const char*,
+                                                     const char*,
+                                                     parse_options_view const&)
+{
+  return cudf::string_view{};
+}
+
+template <>
+__inline__ __device__ cudf::dictionary32 decode_value(const char*,
+                                                      const char*,
+                                                      parse_options_view const&)
+{
+  return cudf::dictionary32{};
+}
+
+template <>
+__inline__ __device__ cudf::list_view decode_value(const char*,
+                                                   const char*,
+                                                   parse_options_view const&)
+{
+  return cudf::list_view{};
+}
+template <>
+__inline__ __device__ cudf::struct_view decode_value(const char*,
+                                                     const char*,
+                                                     parse_options_view const&)
+{
+  return cudf::struct_view{};
+}
+
+template <>
+__inline__ __device__ numeric::decimal32 decode_value(const char*,
+                                                      const char*,
+                                                      parse_options_view const&)
 {
-  return thrust::find_if(thrust::seq, begin, end, [](auto elem) { return elem != ' '; });
+  return numeric::decimal32{};
 }
 
+template <>
+__inline__ __device__ numeric::decimal64 decode_value(const char*,
+                                                      const char*,
+                                                      parse_options_view const&)
+{
+  return numeric::decimal64{};
+}
+
+template <>
+__inline__ __device__ numeric::decimal128 decode_value(const char*,
+                                                       const char*,
+                                                       parse_options_view const&)
+{
+  return numeric::decimal128{};
+}
+
+struct ConvertFunctor {
+  /**
+   * @brief Dispatch for numeric types whose values can be convertible to
+   * 0 or 1 to represent boolean false/true, based upon checking against a
+   * true/false values list.
+   *
+   * @return bool Whether the parsed value is valid.
+   */
+  template <typename T,
+            std::enable_if_t<std::is_integral_v<T> and !std::is_same_v<T, bool> and
+                             !cudf::is_fixed_point<T>()>* = nullptr>
+  __host__ __device__ __forceinline__ bool operator()(char const* begin,
+                                                      char const* end,
+                                                      void* out_buffer,
+                                                      size_t row,
+                                                      const data_type output_type,
```

Comment:
Inconsistency with east/west const in this list. Applies in a few places.
```suggestion
                                                      data_type const output_type,
```

Reply:
Moved (hopefully) all remaining west consts to east const.